testsuite/shared/src/main/java/org/wildfly/test/mixed/domain/TestParserUtils.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.test.mixed.domain;
+
+import javax.xml.namespace.QName;
+
+import org.jboss.as.controller.Extension;
+import org.jboss.as.controller.ProcessType;
+import org.jboss.as.controller.RunningMode;
+import org.jboss.as.controller.RunningModeControl;
+import org.jboss.as.controller.extension.ExtensionRegistry;
+import org.jboss.as.controller.extension.RuntimeHostControllerInfoAccessor;
+import org.jboss.as.subsystem.test.TestParser;
+import org.jboss.staxmapper.XMLMapper;
+
+/**
+ * Builder to create a test xml mapper in full WildFly without depending on the private ExtensionRegistry constructor.
+ * @author Kabir Khan
+ */
+public class TestParserUtils {
+
+    private final XMLMapper xmlMapper;
+    private final String wrappedXml;
+
+    private TestParserUtils(XMLMapper xmlMapper, String wrappedXml) {
+        this.xmlMapper = xmlMapper;
+        this.wrappedXml = wrappedXml;
+    }
+
+    public XMLMapper getXmlMapper() {
+        return xmlMapper;
+    }
+
+    public String getWrappedXml() {
+        return wrappedXml;
+    }
+
+    public static class Builder {
+        private final Extension extension;
+        private final String subsystemName;
+        private final String subsystemXml;
+        private String namespace = "urn.org.jboss.test:1.0";
+        private ProcessType processType = ProcessType.HOST_CONTROLLER;
+        private RunningModeControl runningModeControl = new RunningModeControl(RunningMode.NORMAL);
+        private RuntimeHostControllerInfoAccessor hostControllerInfoAccessor = RuntimeHostControllerInfoAccessor.SERVER;
+
+        public Builder(Extension extension, String subsystemName, String subsystemXml) {
+            this.extension = extension;
+            this.subsystemName = subsystemName;
+            this.subsystemXml = subsystemXml;
+        }
+
+        public Builder setNamespace(String namespace) {
+            this.namespace = namespace;
+            return this;
+        }
+
+        public Builder setProcessType(ProcessType processType) {
+            this.processType = processType;
+            return this;
+        }
+
+        public Builder setRunningModeControl(RunningModeControl runningModeControl) {
+            this.runningModeControl = runningModeControl;
+            return this;
+        }
+
+        public Builder setHostControllerInfoAccessor(RuntimeHostControllerInfoAccessor hostControllerInfoAccessor) {
```

Comment:
This is another class that shouldn't be used in full.

I think this could be a simple setMasterHostController(Boolean master) and then build creates the RuntimeHostControllerInfoAccessor (with null meaning use the SERVER constant.) Or something like that. ;)